CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
+have many parameters, which tend to show up in the SDKs. These are abstracted
+and inferred, where possible, with simple JSON that the implementation can
+reason about. Standard parameters (bucket names, stream names, partition keys,
+etc.) that are part of the implementation are extracted from a simplified
+configuration file and dynamically generated/updated in order to further allow
+focus on state and data types only.
+
+### Amplify Design
+As more and more modules are introduced to AWS Amplify, there is a clear
+necessity to modularize the library into smaller pieces so that users
+could avoid importing unnecessary parts into their app. The goal of this
+design is to make AWS Amplify modularized and also keep it backward
+compatible to avoid breaking changes.
+
+Modular import prevents unnecessary code dependencies are included with the
+app, and thus decreases the bundle size and enables adding new functionality
+without the risk of introducing errors related to the unused code.
+
+Amplify has established the concepts of categories and plugins. A
+__category__ is a collection of api calls that are exposed to the client
+to do things inside that category. For example, in the storage category
+generally one wants to upload and download objects from storage so the
+apis exposed to the client will represent that functionality. Because
+Amplify is pluggable, a __plugin__ of your choosing will provide the
+actual implementation behind that api interface.  Using the same example
+of Storage, the plugin we choose might be `AWSS3StoragePlugin`, which
+would then implement each api call from the category with a service call
+or set of service calls to S3, the underlying storage provider of the
+AWS plugin.
+
+## Getting Started
+
+First, ensure that you have installed the latest stable version of Android
+Studio / the Android SDK.
+
+Configure your environment, so that the `ANDROID_HOME` and `JAVA_HOME`
+environment variables are set. A convenient way of doing this is to add them
+into `~/.bashrc`. On a Mac, the SDK and Java installation used by the SDK may
+be found:
+
+```
+export ANDROID_HOME=~/Library/Android/sdk
+export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home
+```
+Note: JDK 11, 12, 13, etc. have known issues and are not supported.
+
+Now, clone the Amplify Android project from GitHub.
+
+```
+git clone git@github.com:aws-amplify/amplify-android.git
+```
+Load this project into Android Studio by selecting File > Open, and choosing
+the root directory of the project (`amplify-android`). Alternately, cd into this
+top-level directory.
+
+In Android Studio, build the project by clicking the Hammer icon, "Make Project
+⌘F9". If working on the command line, you can do the same thing via:
+
+```
+./gradlew build
+```
+
+## Tools
+[Gradle](https://gradle.org) is used for all [build and dependency management](https://developer.android.com/studio/build).
```

Comment:
It seems strange this bit is after the above build incantation. Perhaps we should introduce the prerequisite tools slightly earlier in the document?

Reply:
The only one of these that needs to be installed into the development environment is Android Studio. (It will bootstrap a couple of other implicit dependencies.) Currently, I have this tools section more like a "code dependencies" section than a "compile time utilities" list. Maybe the easiest fix would be to change the subject heading to something more all-encompassing.